Simbody/Visualizer/src/Visualizer.cpp

```diff
@@ -1107,6 +1108,52 @@ const Array_<Visualizer::FrameController*>& Visualizer::getFrameControllers() co
 const MultibodySystem& Visualizer::getSystem() const {return getImpl().m_system;}
 
 
+//============================================================================
+//                             BODY FOLLOWER
+//============================================================================
+Visualizer::BodyFollower::BodyFollower(
+        const MobilizedBody& body,
+        const Vec3&          stationPinB,
+        const Vec3&          offset,
+        const Vec3&          upDirection)
+    :   m_body(body), m_stationPinB(stationPinB), m_offset(offset),
+        m_upDirection(upDirection) {}
+    
+void Visualizer::BodyFollower::generateControls(
+        const Visualizer&             viz,
+        const State&                  state,
+        Array_< DecorativeGeometry >& geometry)
+{
+    // Offset.
+    Vec3 offset(m_offset);
+    if (m_offset.isNaN()) {
+        // Default: offset is based on system up direction and ground height.
+        offset = Vec3(1, 1, 1);
+        offset[viz.getSystemUpDirection().getAxis()] += viz.getGroundHeight();
+    }
+
+    // Up direction.
+    const Vec3* upDirection;
+    if (m_upDirection.isNaN()) {
+        // Default: use system up direction.
+        upDirection = &UnitVec3(viz.getSystemUpDirection()).asVec3();
+    } else {
+        upDirection = &m_upDirection;
+    }
```

Comment:
> A reference is always preferable if you can do it.

Done.



> One further thought: can you make m_upDirection a UnitVec3? I think it is always a unit vector.

I tried this, but ran into an issue: I get a segmentation fault when I initialize the UnitVec3 to NaN's. So I just went with Vec3's. I think using a Vec3 also makes the constructor easier to use.